Maple2.File.IO/Nif/NifDocument.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System.IO;
+using System.Text;
+
+namespace Maple2.File.IO.Nif;
+
+public class NifDocument {
+    private readonly byte[] fileData;
+    private NifDocumentHeader header;
+    public EndianReader Reader { get; private set; }
+    public string RelPath { get; init; }
+    private NifBlock?[] blocks;
```

Comment:
i don't think you actually want this nullable right?

Reply:
Yeah that was leftover from before I made unimplemented blocks use NifBlock